CATS/Problem/Parser.pm

```diff
@@ -239,36 +242,17 @@ sub inc_object_ref_count {
 sub on_start_tag {
     my CATS::Problem::Parser $self = shift;
     my ($p, $el, %atts) = @_;
-
     my $h = tag_handlers()->{$el};
     my $top_tag = $self->current_tag;
-    if (my $stml = $top_tag ? $top_tag->{stml} : undef) {
+    my $stml = $top_tag ? $top_tag->{stml} : undef;
+    if ($stml && (!$h || !$h->{in_stml})) {
         $h and $self->error("Unexpected top-level tag $el inside stml of " . $self->current_tag->{el});
-        if ($el eq 'include') {
-            my $name = $atts{src} or
-                return $self->error(q~Missing required 'src' attribute of 'include' tag~);
-            $$stml .= Encode::decode(
-                $self->{problem}{encoding}, $self->{source}->read_member($name, "Invalid 'include' reference: '$name'"));
-            return;
-        }
-        elsif ($el eq 'Quiz') {
-            $self->{max_points_quiz} = ($self->{max_points_quiz} // 0) + ($atts{points} // 1);
-            $self->{has_quizzes} = 1;
-        }
-        $$stml .=
-            "<$el" . join ('', map qq~ $_="$atts{$_}"~, keys %atts) . '>';
-
-        $el ne 'img' || $atts{picture} or
-            $self->error('Picture not defined in img element');
-
-        my $attr = { img => 'picture', a => 'attachment', object => 'attachment' }->{$el};
-        $self->inc_object_ref_count($atts{$attr}, $attr) if $attr;
+        $$stml .= "<$el" . join ('', map qq~ $_="$atts{$_}"~, keys %atts) . '>';
```

Comment:
Extract into function (it is repeated below), add `sort` for testability, add test(s)